File: homework/hw_10/p01.cc
#include "lib.h"

void power_loop(int x, int y){
    int product = 1;
    for (int i=0; i<y; i++){
        product = product * x;
    }

    printf("For Loop: (%d ** %d) = %d\n", x, y, product);
}

void recursion(int x, int y, int tmp, int count){
    if (y == count){
        printf("Recursion: (%d ** %d) = %d\n", x, y, tmp);
        exit(0);
    }
    else {
        tmp = tmp * x;
        count ++;
        recursion(x, y, tmp, count);
    }
}

int main (int argc, char ** argv){
    if (argc != 3){
        fprintf(stderr, "Usage: %s <const> <const>\n", argv[0]);
        return 1;
    }

    int x = atoi(argv[1]);
    int y = atoi(argv[2]);

    power_loop(x, y);
    recursion(x, y, 1, 0);
}

File: homework/hw_10/p01.py
#!/usr/bin/env python
import sys

def power_loop(x:int, y:int):
    tmp = 1
    
    for power in range(y):
        tmp = tmp * x
    print("For Loop: pow({} ** {}) = {}".format(x,y,tmp))

def recursion(x:int, y:int, tmp:int, count:int):
    if y == count:
        print("Recursion: pow({} ** {}) = {}".format(x,y,tmp))
        exit()
    else:
        tmp = tmp * x
        count += 1
        recursion(x, y, tmp, count)


def main (argv):
    
    if len(argv) != 3:
        print("Usage: {} <const> <const>".format(argv[0]))
        return 1
    
    power_loop(int(argv[1]), int(argv[2]))
    recursion(int(argv[1]), int(argv[2]), 1, 0)

if __name__ == "__main__":
    main(sys.argv[:])